Live demo playbook

SebastianSchildt opened the pull request:
Notes for live demos

SebastianSchildt committed:
Live demo playbook

Signed-off-by: Sebastian Schildt <sebastian.schildt@de.bosch.com>

eriksven commented on kuksademo/playbook.txt:
```diff
@@ -0,0 +1,53 @@
+# This is my cheat sheet VSS/KUKSA intro
```
"for a"

eriksven commented on kuksademo/playbook.txt:
```diff
@@ -0,0 +1,53 @@
+ pipenv shell
+ pip install kuksa-client
+
+ python data-provider.py
```
For me this script fails with the error below. I will have try to spot the cause for this error myself. 

```
python data-provider.py
Choose a brake pad (FL, FR, RL, RR): FL
Choose a value (0-100): 43
Updating Vehicle.Chassis.Axle.Row1.Wheel.Left.Brake.PadWear to: 43
Traceback (most recent call last):
  File "/Users/jes1be/Documents/SDV/GIT/kuksa/kuksa-incubation/kuksademo/data-provider.py", line 59, in <module>
    main()
    ~~~~^^
  File "/Users/jes1be/Documents/SDV/GIT/kuksa/kuksa-incubation/kuksademo/data-provider.py", line 55, in main
    updateValue(client, path, value)
    ~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jes1be/Documents/SDV/GIT/kuksa/kuksa-incubation/kuksademo/data-provider.py", line 14, in updateValue
    client.set_current_values(
    ~~~~~~~~~~~~~~~~~~~~~~~~~^
        {path: Datapoint(value)})
        ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/jes1be/.local/share/virtualenvs/kuksa-incubation-i5Yue3_e/lib/python3.14/site-packages/kuksa_client/grpc/__init__.py", line 961, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/jes1be/.local/share/virtualenvs/kuksa-incubation-i5Yue3_e/lib/python3.14/site-packages/kuksa_client/grpc/__init__.py", line 1092, in set_current_values
    self.set(
    ~~~~~~~~^
        updates=[
        ^^^^^^^^^
    ...<4 lines>...
        **rpc_kwargs,
        ^^^^^^^^^^^^^
    )
    ^
  File "/Users/jes1be/.local/share/virtualenvs/kuksa-incubation-i5Yue3_e/lib/python3.14/site-packages/kuksa_client/grpc/__init__.py", line 961, in wrapper
    return func(self, *args, **kwargs)
  File "/Users/jes1be/.local/share/virtualenvs/kuksa-incubation-i5Yue3_e/lib/python3.14/site-packages/kuksa_client/grpc/__init__.py", line 1282, in set
    resp = self.client_stub_v2.PublishValueRequest(req, **rpc_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'VALStub' object has no attribute 'PublishValueRequest'
```

eriksven replied:
The error seems to be that I used kuksa-client:0.5.0 which was the latest version at that point in time. In the meantime, kuksa-client:0.5.1 has been released and works for this script.

eriksven commented on kuksademo/playbook.txt:
```diff
@@ -0,0 +1,53 @@
+
+
+
+  getValue Vehicle.Chassis.Axle.*.Wheel.*.Brake.PadWear
```
Is this still to be executed as part of the Python command before? I guess you are supposed to enter this command in the terminal where the kuska-client-cli is running, right? 
If that is the case, we should make this more clear in the sheet cheat.

eriksven commented on kuksademo/driver-ui/webapp/com.js:
```diff
@@ -0,0 +1,154 @@
+}
+
+function tempTostr(wear) {
+    console.log("Wear is "+wear);
```
We should catch null here. Especially, for usability it is a bit confusing to see a green indicator in the UI while the value is displayed as "null". How about, simply showing a ❓ instead?

eriksven commented on kuksademo/data-provider.py:
```diff
@@ -0,0 +1,59 @@
+def main():
+    # Create a VSSClient instance, use 55555 when not on MAC
+
+    client = VSSClient('127.0.0.1', 55556)
```
We could make this configurable as command line parameter. In any case, I would prefer to add a hint in the playbook.txt that the default and the "Mac-port" differ here as well.

eriksven committed:
adds modification to playbook after review

playbook.txt extended
port configurable in data-provider.py
webui display ? N/A if no value is set

SebastianSchildt reviewed (commented):
Thx. lgtm 🚀

bs-jokri reviewed (approved):
LGTM
